src/query/functions/src/scalars/tuple.rs

```diff
@@ -124,32 +126,45 @@ pub fn register(registry: &mut FunctionRegistry) {
             signature: FunctionSignature {
                 name: "get".to_string(),
                 args_type: vec![DataType::Nullable(Box::new(DataType::Tuple(
-                    fields_ty.to_vec(),
+                    (0..fields_ty.len())
+                        .map(|idx| DataType::Nullable(Box::new(DataType::Generic(idx))))
+                        .collect(),
                 )))],
-                return_type: DataType::Nullable(Box::new(fields_ty[idx].clone())),
+                return_type: DataType::Nullable(Box::new(DataType::Generic(idx))),
                 property: FunctionProperty::default(),
             },
             calc_domain: Box::new(move |args_domain| {
                 let NullableDomain { has_null, value } = args_domain[0].as_nullable().unwrap();
-                let value = value.as_ref().map(|value| {
-                    let fields = value.as_tuple().unwrap();
-                    Box::new(fields[idx].clone())
-                });
-                FunctionDomain::Domain(Domain::Nullable(NullableDomain {
-                    has_null: *has_null,
-                    value,
-                }))
+                match value {
+                    Some(value) => {
+                        let NullableDomain {
+                            has_null: field_has_null,
+                            value: field_value,
+                        } = value.as_tuple().unwrap()[idx].as_nullable().unwrap();
+                        FunctionDomain::Domain(Domain::Nullable(NullableDomain {
+                            has_null: *has_null || *field_has_null,
+                            value: field_value.clone(),
+                        }))
+                    }
+                    None => FunctionDomain::Domain(Domain::Nullable(NullableDomain {
+                        has_null: true,
+                        value: None,
+                    })),
+                }
             }),
             eval: Box::new(move |args, _| match &args[0] {
                 ValueRef::Scalar(ScalarRef::Null) => Value::Scalar(Scalar::Null),
                 ValueRef::Scalar(ScalarRef::Tuple(fields)) => Value::Scalar(fields[idx].to_owned()),
                 ValueRef::Column(Column::Nullable(box NullableColumn {
                     column: Column::Tuple { fields, .. },
                     validity,
-                })) => Value::Column(Column::Nullable(Box::new(NullableColumn {
-                    column: fields[idx].to_owned(),
-                    validity: validity.clone(),
-                }))),
+                })) => {
+                    let field_col = fields[idx].as_nullable().unwrap();
+                    Value::Column(Column::Nullable(Box::new(NullableColumn {
+                        column: field_col.column.clone(),
```

Comment:
Is there any way to check the double-nested nullable datatypes during compile?

Reply:
We can check it at type_check.